disnake/guild.py

```diff
@@ -3583,20 +3604,66 @@ async def ban(
         ----------
         user: :class:`abc.Snowflake`
             The user to ban from the guild.
+        clean_history_duration: Union[:class:`int`, :class:`datetime.timedelta`]
+            The timespan (seconds or timedelta) of messages to delete from the user
+            in the guild, up to 7 days (604800 seconds).
+            Defaults to 1 day (86400 seconds).
+
+            This is incompatible with ``delete_message_days``.
+
+            .. versionadded:: 2.6
+
+            .. note::
+                This may not be accurate with small durations (e.g. a few minutes)
+                and delete a couple minutes' worth of messages more than specified.
+
         delete_message_days: :class:`int`
             The number of days worth of messages to delete from the user
             in the guild. The minimum is 0 and the maximum is 7.
+
+            This is incompatible with ``clean_history_duration``.
+
+            .. deprecated:: 2.6
+                Use ``clean_history_duration`` instead.
+
         reason: Optional[:class:`str`]
             The reason for banning this user. Shows up on the audit log.
 
         Raises
         ------
+        TypeError
+            ``clean_history_duration`` has an invalid type.
         Forbidden
             You do not have the proper permissions to ban.
         HTTPException
             Banning failed.
         """
-        await self._state.http.ban(user.id, self.id, delete_message_days, reason=reason)
+        if not (delete_message_days is MISSING) ^ (clean_history_duration is MISSING):
+            raise ValueError(
+                "Exactly one of `clean_history_duration` and `delete_message_days` must be provided."
+            )
```

Comment:
This is a breaking change. If neither are provided, we should default to 1 day, as per the previous implementation, unless we'd like to make this breaking

Reply:
https://github.com/DisnakeDev/disnake/pull/659/commits/274f00986dc7c98fdf2aa2d9fd0bad25b401dd8d

Reply:
Didn't notice, but this is a deviation from our current standard. We typically use TypeError for when mutliple arguments are provided and only one of them can be provided. I haven't seen us use ValueError before, why is this?

Reply:
Good point, fixed in https://github.com/DisnakeDev/disnake/pull/659/commits/4ce431c5e269b4ee06a8c786e7f83ce4ccdccdcf.